appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -236,7 +236,7 @@ else if(backupClassLoader != null) {
             transactionManager.clearThreadTx();
         }
         
-        if (requestTracing != null && requestTracing.isRequestTracingEnabled()) {
+        if (invocation != null && requestTracing != null && requestTracing.isRequestTracingEnabled()) {
```

Comment:
As identified by sonar `invocation` could be null according to other logic in the same method. Only option I saw is to skip calling `startConcurrentContextSpan` in case `invocation` s null as this only works for non-null invocations. So NPE becomes NOOP.

Reply:
This seems like a valid change, but I worry how this will affect the call to `endTrace` in the `reset` method.
As far as I can tell this is an "entry point" for tracing spans on async threads, and as such it's not immediately apparent to me how this might tracing further down the line (assuming execution actually somehow proceeds without having an invocation...).

It might be safer to simply move the null check to the `startConcurrentContextSpan` method and move the creation of the Span tags etc. to inside of that check, so that when the invocation proceeds there's a least a "skeleton" span.